### PR TITLE
Added documentation on how to connect to Google Analytics

### LINKED ISSUE #4812: when adding google analytics as source, it's not clear that I have to enable the API
I spent abount half an hour trying to figure out why I got this strange `closure lang something..` error when adding the source, and the reason was simple: I needed to enable the google analytics API in the google api dashboard.

Also, after understanding that, I got another error.. I was using the tracking id (UA-....) instead of the account id (the tracking id is much more often used when dealing with google analytics)

While this seems obvious now, I would suggest to put small hints to guide users that are not experts in Google APIs, in line with the exceptional onboarding experience of metabase.

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

### PR DESCRIPTION
**Pull request by lama-ank**
###### Before submitting the PR, please make sure you do the following
- [ ] If you're attempting to fix a translation issue, please submit your changes to our [POEditor project](https://poeditor.com/join/project/ynjQmwSsGh) instead of opening a PR.
### Tests
-  [ ] Run the frontend and integration tests with  `yarn lint && yarn flow && yarn test`)
-  [ ] If there are changes to the backend codebase, run the backend tests with `lein test && lein lint && ./bin/reflection-linter`

-  [x] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
(unless it's a tiny documentation change).

### CODE REVIEW

**Comment by lama-ank** [comment]
@salsakran @kdoh I'd like to contribute to Metabase with this document.

After reading the contributing guidelines, I sent this PR but wasn't able to add reviewers to it. 
Would you please point me in the right direction/people I should ask for help to get this PR reviewed?

Thank you in advance

**Comment by nemanjaglumac** [comment]
The remaining failing Cypress test was fixed in `master` and in a `release` branch. There is no need to rerun this whole suite, unless it is rebased on the latest `master`. Since every other check passed, it is safe to merge as it is.

**Comment by nemanjaglumac** [comment]
@ariya you had one PR where PNGs were compressed. PNG included in this commit is little bit over 100kb. Can you share the tool used for compression, if there's even a need for it in this case?

Thanks

**Comment by nemanjaglumac** [comment]
Failing Cypress tests were fixed in both `master` and `release` branches. They can be safely ignored for this PR, especially because it only ever touches `*.md` files.

As soon as @jeff-bruemmer approves this, it is good to merge.

File: docs/administration-guide/databases/google-analytics.md
Comment (dacort) [inline docs/administration-guide/databases/google-analytics.md:72]:
```suggestion
Give Metabase some time to sync with your Google Analytics dataset, then exit Admin mode, click on **Browse Data**, find your database, and start exploring your data. Once Metabase is finished syncing, you should see the names of your Properties & Apps in the data browser. 
```

File: docs/administration-guide/databases/google-analytics.md
Comment (dacort) [inline docs/administration-guide/databases/google-analytics.md:62]:
```suggestion
By default, Metabase does a lightweight hourly sync and an intensive daily scan of field values. If you have a large database, we recommend turning this on and reviewing when and how often the field value scans happen.

See [database sync and analysis documentation](https://www.metabase.com/docs/latest/administration-guide/01-managing-databases.html#database-sync-and-analysis) for more details about these configuration items.
```

**Review by dacort — APPROVED** [review]
Suggested a couple minor changes.

File: docs/administration-guide/databases/google-analytics.md
Comment (flamber) [reply docs/administration-guide/databases/google-analytics.md:72]: GA is actually not synced, since all values are known and pre-defined, so they should be instantly available.

Comment (dacort) [reply docs/administration-guide/databases/google-analytics.md:72]: Odd - when I did this locally, there was a period of maybe a minute where it was showing IDs instead of names and I was getting an error when attempting to query GA.

Comment (flamber) [reply docs/administration-guide/databases/google-analytics.md:72]: Haven't tested for a while, but maybe something else is happening, or maybe I'm misunderstanding the sync+fingerprint code.
It still takes a while to scan, so that would be understandable it wasn't showing names until that was completed.

Comment (lama-ank) [reply docs/administration-guide/databases/google-analytics.md:72]: Hi,thank you everyone for the suggestions and comments. Would it be ok then to say something like "wait for the initial scan to complete" instead of explaining the sync process?

File: docs/administration-guide/databases/google-analytics.md
Comment (jeff-bruemmer) [inline docs/administration-guide/databases/google-analytics.md:13]:
```suggestion
## Google Cloud Platform: creating OAuth client ID

To get the **Client ID** and **Client Secret**, follow Google's Cloud Platform Help on [how to create an OAuth client ID](https://support.google.com/cloud/answer/6158849?hl=en). For application type, select "Desktop App" (this avoids a `redirect_uri_mismatch` when requesting **Auth Codes**).
```

File: docs/administration-guide/databases/google-analytics.md
Comment (jeff-bruemmer) [inline docs/administration-guide/databases/google-analytics.md:17]:
```suggestion
### Enabling Google Analytics API
To enable the Google Analytics API, see [Enable and disable APIs](https://support.google.com/googleapi/answer/6158841?hl=en).
```
We might as well defer to their docs here, so we don't have to worry about the procedure changing

File: docs/administration-guide/databases/google-analytics.md
Comment (jeff-bruemmer) [inline docs/administration-guide/databases/google-analytics.md:22]: Currently Metabase only supports v3. See https://github.com/metabase/metabase/issues/12905

File: docs/administration-guide/databases/google-analytics.md
Comment (jeff-bruemmer) [inline docs/administration-guide/databases/google-analytics.md:27]:
```suggestion
## Metabase: adding a Google Analytics Dataset

In your Metabase, click on the **settings** cog, and select **Admin** to bring up the Admin panel. In the **Databases** section, click on the **Add database** button in the upper right.
```

File: docs/administration-guide/databases/google-analytics.md
Comment (jeff-bruemmer) [inline docs/administration-guide/databases/google-analytics.md:50]:
```suggestion
Once you've provided **Client ID** and **Client Secret** with valid scopes, a **Click here to get an auth code** link will appear over the **Auth Code** text box. Authorize the connection with your Google Login Credentials to see the Auth Code, then copy and paste the code here.
```

**Review by jeff-bruemmer — CHANGES_REQUESTED** [review]
@lama-ank, I've added some suggestions for revision. 

Thanks for the PR! Cheers.

**Comment by lama-ank** [comment]
> @lama-ank, I've added some suggestions for revision.
> 
> Thanks for the PR! Cheers.

My pleasure! Thanks for reviewing and proposing suggestions.

File: docs/administration-guide/databases/google-analytics.md
Comment (lama-ank) [reply docs/administration-guide/databases/google-analytics.md]: Removed reference to Google Analytics versions for API scopes.

File: docs/administration-guide/databases/google-analytics.md
Comment (etoulas) [inline docs/administration-guide/databases/google-analytics.md:14]: I just followed your instructions and had some difficulties at this step. Until I actually added the database in Metabase and came cross this help text with the _correct_ link to enable the GA API.

> To use Metabase with this data you must enable API access in the Google Developers Console. [Click here](https://console.developers.google.com/apis/api/analytics.googleapis.com/overview) to go to the console if you haven't already done so.

image

Comment (etoulas) [reply docs/administration-guide/databases/google-analytics.md:14]:
```suggestion
To enable the Google Analytics API, go to this page in the Google Cloud Platform console: https://console.cloud.google.com/apis/api/analytics.googleapis.com/overview. Double check that the previously created project is selected and click on "Enable".

For further documentation please refer to [Enable and disable APIs](https://support.google.com/googleapi/answer/6158841?hl=en).
```

**Review by etoulas — CHANGES_REQUESTED** [review]

File: docs/administration-guide/databases/google-analytics.md
Comment (lama-ank) [reply docs/administration-guide/databases/google-analytics.md]: I see your point. This certainly improves the flow when following the instructions. Thanks for performing this correction.

**Comment by lama-ank** [comment]
> > @lama-ank, I've added some suggestions for revision.
> > Thanks for the PR! Cheers.
> 
> My pleasure! Thanks for reviewing and proposing suggestions.

Hi, I was wondering if you had the chance to check back this PR. 
Thanks

**Review by jeff-bruemmer — APPROVED** [review]

**Comment by lama-ank** [comment]
Hi, I think all is done now.  Thanks for your collaboration.

**Comment by lama-ank** [comment]
Since this one is approved, would you please merge it?
Thanks

**Comment by paoliniluis** [comment]
hey! as it's a 3rd party contribution, tests didn't pass on our CI. I'll ask for someone to force merge it or I'll cherry pick it on a personal branch and merge it. Thanks for the contribution BTW

**Comment by flamber** [comment]
There's an updated version of this here: #17693

**Comment by jeff-bruemmer** [comment]
Closing this, as #17693 has been merged. Thanks, again, @lama-ank your changes were applied to the other PR.

**Comment by lama-ank** [comment]
It's been a pleasure.

Regards

**Pull request closed without merging**
